tests/integration-tests/tests/proxy/test_proxy.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import boto3
+import pytest
+from assertpy import assert_that
+from cfn_stacks_factory import CfnStack
+from remote_command_executor import RemoteCommandExecutor
+from utils import generate_stack_name, get_username_for_os
+
+
+@pytest.fixture(scope="class")
+def proxy_stack(region, request, cfn_stacks_factory):
+    """
+    Set up and tear down a CloudFormation stack to deploy a proxy environment.
+
+    The stack is created from a CloudFormation template that sets up the necessary
+    networking, instances, and security groups for a proxy environment. The proxy
+    address and subnet are retrieved from the stack outputs and used in the cluster
+    configuration.
+    """
+    proxy_stack_template_path = "../../cloudformation/proxy/proxy.yaml"
+    with open(proxy_stack_template_path) as proxy_stack_template:
+        stack_name = generate_stack_name("integ-tests-proxy", request.config.getoption("stackname_suffix"))
```

Comment:
As a stretch goal (can be done also in a follow up PR) let's implement the possibility to reuse an existing Proxy stack, so that the test do not need to recreate it every time you run the test.

See how it has been implement for the VPC stack.

Reply:
Done! Now we can reuse an existing Proxy Stack by specify `--proxy-stack PROXY_STACK_NAME` when run the test.